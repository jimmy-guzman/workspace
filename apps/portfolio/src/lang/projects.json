{
  "projects": [
    {
      "name": "Recipe Box",
      "repo": "recipebox",
      "url": "https://recipebox.jimmyguzman.com/",
      "description": "This Recipe Box was built with React, Redux and Sass. It's a webapp where the user is able to view, create and edit recipes.",
      "spanish": "Esta Recipe Box fue construida con React, Redux y Sass. Es una aplicación de web en la que el usuario puede ver, crear y editar recetas.",
      "tech": ["ReactJS", "Redux", "JavaScript", "Sass", "JSX", "Parcel"]
    },

    {
      "name": "Markdown Previewer",
      "repo": "markdownpreviewer",
      "url": "https://jimmy-guzman.github.io/markdownpreviewer/",
      "description": "The Markdown Previewer is a webapp that incorporates both React and Sass to render a preview of markdown in real time.",
      "spanish": "El Markdown Previewer es una aplicación de web que incorpora React y Sass para renderizar Markdown en tiempo real.",
      "tech": ["ReactJS", "Sass", "marked.js"]
    },
    {
      "name": "Tic Tac Toe Game",
      "repo": "tictactoegame",
      "url": "https://jimmy-guzman.github.io/tictactoegame/",
      "description": "In this Tic Tac Toe Game, the user is able to play tic tac toe versus the computer. The computer ai is based on the minimax algorithm.",
      "spanish": "En este juego de Tic Tac Toe, el usuario puede jugar tres en raya contra la computadora. El A.I. del la computadora es basado en el algoritmo minimax.",
      "tech": ["JavaScript", "HTML", "CSS"]
    },
    {
      "repo": "pomodoroclock",
      "name": "Pomodoro Clock",
      "url": "https://jimmy-guzman.github.io/pomodoroclock/",
      "description": "This Pomodoro Clock is a web app that acts as minimalist Pomodoro Technique timer with options for user ease.",
      "spanish": "Este Pomodoro Clock es una aplicación de web que actúa como un temporizador minimalista de Pomodoro Technique con opciones para la facilidad del usuario.",
      "tech": ["JavaScript", "HTML", "CSS"]
    },
    {
      "repo": "javascriptcalculator",
      "name": "Javascript Calculator",
      "url": "https://jimmy-guzman.github.io/javascriptcalculator/",
      "description": "The Javascript Calculator is a web app that replicates a basic calculator through Javascript.",
      "spanish": "La Calculadora de Javascript es una aplicación se web que replica una calculadora básica a través de Javascript.",
      "tech": ["JavaScript", "HTML", "CSS"]
    },
    {
      "repo": "employeedirectorydashboard",
      "name": "Employee Directory Dashboard",
      "url": "https://jimmy-guzman.github.io/employeedirectorydashboard/",
      "description": "Employee directory dashboard that communicates with a third-party API.",
      "spanish": "Tablero de directorios de empleados que se comunica con una API de terceros",
      "tech": ["JavaScript", "jQuery", "HTML", "CSS"]
    },
    {
      "repo": "webappdashboard",
      "name": "Web App Dashboard",
      "url": "https://jimmy-guzman.github.io/webappdashboard/",
      "description": "Web dashboard complete with JavaScript-driven charts and graphs.",
      "spanish": "Tablero web completo con tablas y gráficos guiados por JavaScript.",
      "tech": ["JavaScript", "ChartJS", "Sass", "HTML"]
    },
    {
      "repo": "wikipediaviewer",
      "name": "Wikipedia Viewer",
      "url": "https://jimmy-guzman.github.io/wikipediaviewer/",
      "description": "The Wikipedia Viewer pulls results or a random article from Wikipedia's API based on the user's criteria.",
      "spanish": "El Wikipedia Viewer extrae resultados o un artículo aleatorio de la API de Wikipedia basado en los criterios del usuario.",
      "tech": ["JavaScript", "HTML", "CSS"]
    }
  ]
}
