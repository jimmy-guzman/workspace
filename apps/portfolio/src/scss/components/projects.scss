.projects {
  @include e(buttons) {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 0;
    @include flex(flex, row, space-between, center);
    @include mq(medium) {
      bottom: 50%;
    }
    @include mq(large) {
      left: 10%;
      right: 10%;
    }
  }
}

.project {
  width: 100%;
  @include mq(medium) {
    max-width: 768px;
    width: 80%;
  }
  margin: 0 auto;
  @include e(image) {
    padding: 0;
    position: relative;
    img {
      display: block;
      width: 100%;
    }
  }
  @include e(bar) {
    background: $color-bg-dark;
    color: $color-text-dark;
    letter-spacing: $letter-spacing-more;
    @include flex(flex, row, space-between, center);
    span {
      padding: em(8px);
      background: $color-bg-base;
    }
  }
  @include e(links) {
    margin-right: em(8px);
    a {
      margin-left: em(16px);
    }
  }
  @include e(info) {
    background: $color-bg-dark;
    padding: em(16px);
  }
  @include e(desc) {
    margin: 0;
  }
}
