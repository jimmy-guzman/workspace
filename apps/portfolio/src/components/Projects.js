import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { projects } from '../data/projects.json'
import Project from './Project'
import ProjectButtons from './ProjectButtons'

class Projects extends Component {
  state = {
    projectIndex: 0,
    loadingProjectImage: true
  }

  renderNext = () => {
    let { projectIndex } = this.state

    if (projectIndex === projects.length - 1) {
      projectIndex = 0
    } else {
      projectIndex += 1
    }

    this.setState({ projectIndex, loadingProjectImage: true })
  }

  renderPrev = () => {
    let { projectIndex } = this.state

    if (projectIndex === 0) {
      projectIndex = projects.length - 1
    } else {
      projectIndex += 1
    }

    this.setState({ projectIndex, loadingProjectImage: true })
  }

  onProjectImageLoaded = () => {
    this.setState({ loadingProjectImage: false })
  }

  render() {
    const { projectIndex, loadingProjectImage } = this.state
    const { isEnglish } = this.props

    return (
      <main className='light'>
        <section className='projects grid__row'>
          <Project
            loadingProjectImage={loadingProjectImage}
            project={projects[projectIndex]}
            isEnglish={isEnglish}
            onProjectImageLoaded={this.onProjectImageLoaded}
          />
          <ProjectButtons
            renderNext={this.renderNext}
            renderPrev={this.renderPrev}
          />
        </section>
      </main>
    )
  }
}

Projects.propTypes = {
  isEnglish: PropTypes.bool.isRequired
}

export default Projects
