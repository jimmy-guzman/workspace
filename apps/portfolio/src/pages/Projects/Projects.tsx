import React, { useState } from 'react'

import { projects as defaultProjects } from '../../lang/projects.json'
import { Project, ProjectButtons } from '../../components'

export interface ProjectsProps {
  isEnglish: boolean
  projects: Project[]
}

export const Projects = ({
  isEnglish,
  projects = defaultProjects
}: ProjectsProps) => {
  const [projectIndex, setProjectIndex] = useState<number>(0)
  const [loadingProjectImage, setLoadingProjectImage] = useState<boolean>(true)

  const renderNext = () => {
    setProjectIndex(prevState => {
      return prevState === projects.length - 1 ? 0 : (prevState += 1)
    })
    setLoadingProjectImage(true)
  }

  const renderPrev = () => {
    setProjectIndex(prevState => {
      if (prevState === projects.length - 1) {
        return 0
      }

      return projectIndex === 0 ? projects.length - 1 : (prevState += 1)
    })
    setLoadingProjectImage(true)
  }

  const onProjectImageLoaded = () => {
    setLoadingProjectImage(false)
  }

  return (
    <main className='light'>
      <section className='projects grid__row'>
        <Project
          loadingProjectImage={loadingProjectImage}
          project={projects[projectIndex]}
          isEnglish={isEnglish}
          onProjectImageLoaded={onProjectImageLoaded}
        />
        <ProjectButtons renderNext={renderNext} renderPrev={renderPrev} />
      </section>
    </main>
  )
}
