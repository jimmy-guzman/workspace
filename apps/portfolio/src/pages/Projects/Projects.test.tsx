import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import { projects as defaultProjects } from '../../lang/projects.json'
import { Projects, ProjectsProps } from './Projects'

const setupProjects = (
  props: ProjectsProps = {
    isEnglish: true,
    projects: defaultProjects,
  }
) => {
  return render(<Projects {...props} />)
}

describe('<Projects />', () => {
  it('should render english snapshot', () => {
    const { container } = setupProjects()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render spanish snapshot', () => {
    const { container } = setupProjects({
      isEnglish: false,
      projects: defaultProjects,
    })

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render next project', () => {
    const { getByLabelText, getByText } = setupProjects()

    fireEvent.click(getByLabelText('Next'))

    expect(getByText(defaultProjects[1].name)).toBeInTheDocument()
  })
  it('should render previous project', () => {
    const { getByLabelText, getByText } = setupProjects()

    fireEvent.click(getByLabelText('Previous'))

    expect(
      getByText(defaultProjects[defaultProjects.length - 1].name)
    ).toBeInTheDocument()
  })
})
