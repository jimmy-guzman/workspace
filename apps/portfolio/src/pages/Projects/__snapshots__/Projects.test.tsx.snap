// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Projects /> should render english snapshot 1`] = `
.c0 {
  margin: 0;
  position: relative;
  height: 8px;
  display: block;
  width: 100%;
  background-color: #292b2f;
  overflow: hidden;
}

.c1 {
  background-color: #018185;
}

.c1:before {
  content: '';
  position: absolute;
  background-color: inherit;
  top: 0;
  left: 0;
  bottom: 0;
  will-change: left,right;
  -webkit-animation: emTRrC 2.1s cubic-bezier(0.65,0.815,0.735,0.395) infinite;
  animation: emTRrC 2.1s cubic-bezier(0.65,0.815,0.735,0.395) infinite;
}

.c1:after {
  content: '';
  position: absolute;
  background-color: inherit;
  top: 0;
  left: 0;
  bottom: 0;
  will-change: left,right;
  -webkit-animation: gbxbMm 2.1s cubic-bezier(0.165,0.84,0.44,1) infinite;
  animation: gbxbMm 2.1s cubic-bezier(0.165,0.84,0.44,1) infinite;
  -webkit-animation-delay: 1.15s;
  animation-delay: 1.15s;
}

<main
  class="light"
>
  <section
    class="projects grid__row"
  >
    <div
      class="project grid__row"
    >
      <div
        class="project__image card"
      >
        <div
          class="project__bar"
        >
          <span>
            Recipe Box
          </span>
          <div
            class="project__links"
          >
            <a
              aria-label="View Demo"
              href="https://recipebox.jimmyguzman.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                height="24"
                viewBox="0 0 448 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              aria-label="View Repo"
              href="https://github.com/jimmy-guzman/recipebox"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                height="24"
                viewBox="0 0 496 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="c0"
        >
          <div
            class="c1"
          />
        </div>
        <img
          alt="Recipe Box"
          src="https://jimmy-guzman.imgix.net/project-screenshots/recipebox.png?w=400?fm=png&auto=format"
          srcset="https://jimmy-guzman.imgix.net/project-screenshots/recipebox.png?w=400?fm=png&auto=format 1x,
        https://jimmy-guzman.imgix.net/project-screenshots/recipebox.png?w=400?fm=png&auto=format&fit=max&q=40&dpr=2 2x,
        https://jimmy-guzman.imgix.net/project-screenshots/recipebox.png?w=400?fm=png&auto=format&fit=max&q=20&dpr=3 3x"
          style="visibility: hidden;"
        />
        <div
          class="project__info"
        >
          <ul
            class="TechList"
            style="text-align: center; list-style: none; padding: 0px;"
          >
            <li
              class="badge"
            >
              ReactJS
            </li>
            <li
              class="badge"
            >
              Redux
            </li>
            <li
              class="badge"
            >
              JavaScript
            </li>
            <li
              class="badge"
            >
              Sass
            </li>
            <li
              class="badge"
            >
              JSX
            </li>
            <li
              class="badge"
            >
              Parcel
            </li>
          </ul>
          <p
            class="project__desc"
          >
            This Recipe Box was built with React, Redux and Sass. It's a webapp where the user is able to view, create and edit recipes.
          </p>
        </div>
      </div>
    </div>
    <div
      class="projects__buttons"
    >
      <button
        aria-label="Previous"
        class="btn__secondary"
        type="button"
      >
        <svg
          height="48"
          viewBox="0 0 448 512"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        aria-label="Next"
        class="btn__secondary"
        type="button"
      >
        <svg
          height="48"
          viewBox="0 0 448 512"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </section>
</main>
`;

exports[`<Projects /> should render spanish snapshot 1`] = `
.c0 {
  margin: 0;
  position: relative;
  height: 8px;
  display: block;
  width: 100%;
  background-color: #292b2f;
  overflow: hidden;
}

.c1 {
  background-color: #018185;
}

.c1:before {
  content: '';
  position: absolute;
  background-color: inherit;
  top: 0;
  left: 0;
  bottom: 0;
  will-change: left,right;
  -webkit-animation: emTRrC 2.1s cubic-bezier(0.65,0.815,0.735,0.395) infinite;
  animation: emTRrC 2.1s cubic-bezier(0.65,0.815,0.735,0.395) infinite;
}

.c1:after {
  content: '';
  position: absolute;
  background-color: inherit;
  top: 0;
  left: 0;
  bottom: 0;
  will-change: left,right;
  -webkit-animation: gbxbMm 2.1s cubic-bezier(0.165,0.84,0.44,1) infinite;
  animation: gbxbMm 2.1s cubic-bezier(0.165,0.84,0.44,1) infinite;
  -webkit-animation-delay: 1.15s;
  animation-delay: 1.15s;
}

<main
  class="light"
>
  <section
    class="projects grid__row"
  >
    <div
      class="project grid__row"
    >
      <div
        class="project__image card"
      >
        <div
          class="project__bar"
        >
          <span>
            Recipe Box
          </span>
          <div
            class="project__links"
          >
            <a
              aria-label="View Demo"
              href="https://recipebox.jimmyguzman.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                height="24"
                viewBox="0 0 448 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z"
                  fill="currentColor"
                />
              </svg>
            </a>
            <a
              aria-label="View Repo"
              href="https://github.com/jimmy-guzman/recipebox"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                height="24"
                viewBox="0 0 496 512"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="c0"
        >
          <div
            class="c1"
          />
        </div>
        <img
          alt="Recipe Box"
          src="https://jimmy-guzman.imgix.net/project-screenshots/recipebox.png?w=400?fm=png&auto=format"
          srcset="https://jimmy-guzman.imgix.net/project-screenshots/recipebox.png?w=400?fm=png&auto=format 1x,
        https://jimmy-guzman.imgix.net/project-screenshots/recipebox.png?w=400?fm=png&auto=format&fit=max&q=40&dpr=2 2x,
        https://jimmy-guzman.imgix.net/project-screenshots/recipebox.png?w=400?fm=png&auto=format&fit=max&q=20&dpr=3 3x"
          style="visibility: hidden;"
        />
        <div
          class="project__info"
        >
          <ul
            class="TechList"
            style="text-align: center; list-style: none; padding: 0px;"
          >
            <li
              class="badge"
            >
              ReactJS
            </li>
            <li
              class="badge"
            >
              Redux
            </li>
            <li
              class="badge"
            >
              JavaScript
            </li>
            <li
              class="badge"
            >
              Sass
            </li>
            <li
              class="badge"
            >
              JSX
            </li>
            <li
              class="badge"
            >
              Parcel
            </li>
          </ul>
          <p
            class="project__desc"
          >
            Esta Recipe Box fue construida con React, Redux y Sass. Es una aplicación de web en la que el usuario puede ver, crear y editar recetas.
          </p>
        </div>
      </div>
    </div>
    <div
      class="projects__buttons"
    >
      <button
        aria-label="Previous"
        class="btn__secondary"
        type="button"
      >
        <svg
          height="48"
          viewBox="0 0 448 512"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        aria-label="Next"
        class="btn__secondary"
        type="button"
      >
        <svg
          height="48"
          viewBox="0 0 448 512"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </section>
</main>
`;
